firmware/library/L1_Drivers/gpio.hpp

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "L0_LowLevel/LPC40xx.h"
+
+class GPIOInterface
+{
+ public:
+     virtual void SetAsInput(void)             = 0;
+     virtual void SetAsOutput(void)            = 0;
+     virtual void SetDir(bool input = true)    = 0;
+     virtual void CtrlOutputHigh(void)         = 0;
+     virtual void CtrlOutputLow(void)          = 0;
+     virtual void CtrlOutput(bool high = true) = 0;
```

Comment:
I think that replacing `Ctrl` with `Set` would be more readable. Ctrl seems kinda confusing. Set was also make more sense in place of `CtrlOutput` especially since it sets the pin to high. So, `SetOutput()` can either mean, set it to the input parameter or set it (the opposite of clear) the pin.

Reply:
Is it alright if I just change the names to SetHigh(), SetLow(), and Set(bool high = true)?  I like those names better because users would not have to write a long method name when using those methods.

Reply:
Thats good!